src/apps/concepts/pages/CreateOrEditConceptPage.tsx

```diff
@@ -113,12 +113,14 @@ const CreateOrEditConceptPage: React.FC<Props> = ({
   useEffect(() => {
     // only retrieve the concept if the context was edit at the beginning
     // otherwise we obviously have nothing to edit
-    if (originallyEditing) retrieveConcept(conceptUrl);
-    retrieveDictionary(linkedDictionaryUrl);
+    return () => { resetConceptForm(); }
```

Comment:
We actually don't want to touch *this* `useEffect()` call or at least we don't want to overwrite it. I suppose what we should have here is this:

```js
 useEffect(() => {
    // only retrieve the concept if the context was edit at the beginning
    // otherwise we obviously have nothing to edit
    if (originallyEditing) retrieveConcept(conceptUrl);
    retrieveDictionary(linkedDictionaryUrl);

    return () => resetConceptForm();

    // usually doing the following is a mistake and will bite us later
    // eslint-disable-next-line react-hooks/exhaustive-deps
}, []);
```

This would replace everything in to commit, as we only need to return the function that calls `resetConceptForm()` once.